contracts/abstracted_account.algo.ts

```diff
@@ -82,12 +89,12 @@ export class AbstractedAccount extends Contract {
   }
 
   /**
-   * Temporarily rekey to an approved plugin app address
+   * Temporarily rekey to an approved plugin app address including factory plugins
    *
    * @param plugin The app to rekey to
    */
   rekeyToPlugin(plugin: Application): void {
-    assert(this.plugins(plugin).exists);
+    assert(this.plugins(plugin).exists || this.factories(plugin).exists);
```

Comment:
Under what circumstances would it need to be rekeyed to a factory? I was thinking factories wouldn't require a rekey

Reply:
my bad, i think i got mixed up here.

The factory would need to use the flash rekey functionality in order to work would it not?

This definitely wont work, because it would give anyone permission to call those factory methods but it still needs to rekey to let the factory add the child plugin to your account right? Just triggered by the EOA or Admin?

Reply:
1. Admin adds factory app to abstracted account. This adds the factory app the abstracted account so it allows plugins to be added from it
1.  Admin calls factory app. This adds the created app to the plugin list

I suppose the key here is that the factory app has a check to ensure the caller is the admin of the abstracted account. 

Now that I think about it though there really isn't much of a benefit from separating factories from plugin. The security assumptions are basically the same because the factory could mint a malicious plugin and then use that plugin.

Reply:
yeah, i was thinking this through and started to question what the point of the factories box was if the admin is required to call the factory in the first place